### PR TITLE
GitIgnore appsettings.json and add appsettings.json.sample

### PR DESCRIPTION
**Pull request by palatter**
# What

This will allow us to have our own appsettings.json with connection string locally without it showing up in git or being accidentally commited.

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: samples/Server/appsettings.json.sample
Comment (JamesBurnside) [inline samples/Server/appsettings.json.sample:1]: To remove the step of having to clone and rename the `.sample`, could we keep this as `appsettings.json` and just include it in the gitgnore?

Comment (palatter) [reply samples/Server/appsettings.json.sample:1]: We could do that but then you would not know what the intended structure of the file should be.

You should only have to do this copy of the file the first time you checkout the repo though.

Comment (palatter) [reply samples/Server/appsettings.json.sample:1]: We could include the complete intended structure of the file in documentation?

Comment (JamesBurnside) [reply samples/Server/appsettings.json.sample:1]: Is a good point, you might think its an irrelevant file and it would be hard to recover

**Review by jpeng-ms — APPROVED** [review]

**Pull request merged**